developer-guides/miscellaneous/frontend-ui-tutorial/wallet-usage.md

```diff
@@ -0,0 +1,300 @@
+---
+description: Follow this tutorial to incorporate CosmosKit wallet selection into your dapp.
+---
+
+# Wallet usage
+
+{% hint style="info" %}
+You can find all the code on the GitHub repository: [https://github.com/topmonks/juno-dapp-tutorial](http://localhost:5000/o/jCrX4yBFt02YOr6hTAgL/s/uwjK0eMfGoo1QBmmL4yh/)
+{% endhint %}
+
+### Installation
+
+If you're starting a new project and prefer a streamlined process, you can use the CosmosKit starter tool called "create-cosmos-app". This tool helps you bootstrap your project with Cosmos-related dependencies in just one command. Check [https://github.com/cosmology-tech/create-cosmos-app](https://github.com/cosmology-tech/create-cosmos-app).&#x20;
+
+We follow the explicit installation process in the tutorial using an already set up React app.
+
+#### CosmosKit
+
+Start by installation of the `@cosmos-kit`
+
+```bash
+npm i @cosmos-kit/react @cosmos-kit/core @cosmos-kit/keplr 
+```
+
+{% hint style="info" %}
+When using Vite.js with CosmosKit, it's important to note that CosmosKit's dependencies may require the use of polyfills. You may encounter errors like `Buffer is not defined` in the browser console. To resolve this issue, you can follow these steps:
+
+1. `npm i -D vite-plugin-node-polyfills`
+2.  update the `vite.config.js` file&#x20;
+
+    ```typescript
+    import { defineConfig } from "vite";
+    import { nodePolyfills } from "vite-plugin-node-polyfills";
+
+    export default defineConfig({
+      plugins: [
+        // ...
+        nodePolyfills({
+          protocolImports: true,
+        }),
+      ],
+    });
+    ```
+{% endhint %}
+
+To ensure you have the most up-to-date Juno chain information managed by the community, install the official `chain-registry`
+
+```bash
+npm i chain-registry
+```
+
+#### UI framework
+
+If you have already included a UI kit in your project, you will need to provide the CosmosKit function to generate the wallet modal. This modal is used for wallet integration and can be customized according to your needs. You can refer to the CosmosKit documentation for more details on how to customize the modal with the walletModal function: [https://docs.cosmoskit.com/provider/chain-provider#customize-modal-with-walletmodal](https://docs.cosmoskit.com/provider/chain-provider#customize-modal-with-walletmodal).
```

Comment:
The link is broken. This seems to be the updated, working one: https://docs.cosmology.zone/cosmos-kit/provider/chain-provider#customize-modal-with-walletmodal